### PR TITLE
Lf 2086 - when creating crop plan, warn user of organic status mismatch between varietal and location

### PR DESCRIPTION
**Pull request by anant00729**
To Test: 

When creating a crop plan warn of mismatches using the following 3 cases:

Case 1: Organic varietal on an organic location → No pop-up
Case 2: Non-organic varietal on a non-organic location → No pop-up
Case 3: Organic varietal on a non-organic location → Pop-up
Case 4: Non-organic varietal on an organic location → Pop-up

Case 3:
Title: “You’ve selected a non-organic location!”
Body: “You’ve indicated you’ll be planting an organic crop in a non-organic location. Do you want to proceed?”
Button: “Go back”

Clicking this button returns the user to the “Select your location” page of the crop plan and unselects the non-organic selection

Button: “That’s fine”
Clicking this button returns the user to the “Select your location” page and maintains the user’s selection

Case 4:
Title: “You’ve selected an organic location!”
Body: “You’ve indicated you’ll be planting a non-organic crop in an organic location. Do you want to proceed?”
Button: “Go back”
Clicking this button returns the user to the “Select your location” page of the crop plan and unselects the organic selection
Button: “That’s fine”
Clicking this button returns the user to the “Select your location” page and maintains the user’s selection

Pertinent edge cases:
For crop plans that will have a transplant task, the mismatch check should be made for each location
Locations that are currently “Transitioning” status should be treated as “Non-organic”

Additional context: 
This pop-up should display ANY time a mismatch occurs.

For more details: 
https://lite-farm.atlassian.net/browse/LF-2086

### CODE REVIEW

File: packages/webapp/src/components/Crop/PlantingLocation/index.jsx [hunk 1]
```diff
@@ -98,11 +100,24 @@ export default function PurePlantingLocation({
...
+  const crop = useSelector(cropVarietySelector(variety_id));
+
+  const onSubmit = () => {
+    const d = getValues();
+    const selectedLocationId =
+      d?.crop_management_plan?.planting_management_plans[locationPrefix]?.location_id;
+    const selectedLocation = cropLocations.find((c) => c.location_id === selectedLocationId);
+    const isCropOrganic = crop.organic;
+    const isSelectedLocationOrganic = selectedLocation.organic_status === 'Organic';
+    if (isCropOrganic !== isSelectedLocationOrganic) {
```
Comment (Duncan-Brain) [inline packages/webapp/src/components/Crop/PlantingLocation/index.jsx:120]: I think the default value for crop.organic could be null in some cases, based the cropVarietyModel in the api. If null the expression is always true. Is there a way to set this property? I haven't found it in the ui.

**Comment by Duncan-Brain** [comment]
I cannot seem to get past the pop-up. After pressing "Thats fine" the flow does not continue.

**Comment by Duncan-Brain** [comment]
> To Test:
> 
> When creating a crop plan warn of mismatches using the following 3 cases:
> 
> Case 1: Organic varietal on an organic location → No pop-up Case 2: Non-organic varietal on a non-organic location → No pop-up Case 3: Organic varietal on a non-organic location → Pop-up Case 4: Non-organic varietal on an organic location → Pop-up
> 
> Case 3: Title: “You’ve selected a non-organic location!” Body: “You’ve indicated you’ll be planting an organic crop in a non-organic location. Do you want to proceed?” Button: “Go back”
> 
> Clicking this button returns the user to the “Select your location” page of the crop plan and unselects the non-organic selection
> 
> Button: “That’s fine” Clicking this button returns the user to the “Select your location” page and maintains the user’s selection
> 
> Case 4: Title: “You’ve selected an organic location!” Body: “You’ve indicated you’ll be planting a non-organic crop in an organic location. Do you want to proceed?” Button: “Go back” Clicking this button returns the user to the “Select your location” page of the crop plan and unselects the organic selection Button: “That’s fine” Clicking this button returns the user to the “Select your location” page and maintains the user’s selection
> 
> Pertinent edge cases: For crop plans that will have a transplant task, the mismatch check should be made for each location Locations that are currently “Transitioning” status should be treated as “Non-organic”
> 
> Additional context: This pop-up should display ANY time a mismatch occurs.
> 
> For more details: https://lite-farm.atlassian.net/browse/LF-2086

**Comment by anant00729** [comment]
In the ticket its mentioned that 

"Clicking this button returns the user to the “Select your location” page and maintains the user’s selection"

This means that the farm area will be selected and the user can select another area/location to proceed to the next step.

**Comment by Duncan-Brain** [comment]
My apologies for accidentally closing this. 

For continuation @anant00729 re-submitted his PR here: https://github.com/LiteFarmOrg/LiteFarm/pull/2301

**Pull request closed without merging**